lib/instrumentation/transaction.js

```diff
@@ -152,7 +175,10 @@ Transaction.prototype.toJSON = function () {
         this._user
       ),
       tags: this._labels || {},
-      custom: this._custom || {}
+      custom: this._custom || {},
+      service: this._service || {},
+      cloud: this._cloud || {},
+      message: this._message || {}
```

Comment:
Actually I think the `|| {}` here is why our encoded transactions have empty objects for unused `context.$subkey`s. I think it would be nicer if those were not included when empty.

Reply:
I had a similar thoughts when adding this but was unsure what effect, if any, not serializing these would have, so I erred on the side of cargo cult-ing the existing behavior.  I've opened a separate issue to investigate what effect (if any) _not_ including the properties when empty would have. https://github.com/elastic/apm-agent-nodejs/issues/2416